geode-core/src/main/java/org/apache/geode/management/internal/cli/functions/DestroyJndiBindingFunction.java

```diff
@@ -21,22 +21,29 @@
 import org.apache.geode.management.cli.CliFunction;
 import org.apache.geode.management.internal.cli.i18n.CliStrings;
 
-public class DestroyJndiBindingFunction extends CliFunction<String> {
-
-  static final String RESULT_MESSAGE = "Jndi binding \"{0}\" destroyed on \"{1}\"";
-  static final String EXCEPTION_RESULT_MESSAGE = "Jndi binding \"{0}\" not found on \"{1}\"";
+public class DestroyJndiBindingFunction extends CliFunction<Object[]> {
 
   @Override
-  public CliFunctionResult executeFunction(FunctionContext context) {
-    String jndiName = (String) context.getArguments();
+  public CliFunctionResult executeFunction(FunctionContext<Object[]> context) {
+    String jndiName = (String) context.getArguments()[0];
+    boolean destroyingDataSource = (boolean) context.getArguments()[1];
+
+    String typeName = "Jndi binding";
+
+    if (destroyingDataSource) {
+      typeName = "Data source";
+    }
+
+    final String RESULT_MESSAGE = "{0} \"{1}\" destroyed on \"{2}\"";
+    final String EXCEPTION_RESULT_MESSAGE = "{0} \"{1}\" not found on \"{2}\"";
 
     try {
       JNDIInvoker.unMapDatasource(jndiName);
-      return new CliFunctionResult(context.getMemberName(), true,
-          CliStrings.format(RESULT_MESSAGE, jndiName, context.getMemberName()));
+      return new CliFunctionResult(context.getMemberName(), CliFunctionResult.StatusState.OK,
+          CliStrings.format(RESULT_MESSAGE, typeName, jndiName, context.getMemberName()));
     } catch (NamingException e) {
-      return new CliFunctionResult(context.getMemberName(), true,
-          CliStrings.format(EXCEPTION_RESULT_MESSAGE, jndiName, context.getMemberName()));
+      return new CliFunctionResult(context.getMemberName(), CliFunctionResult.StatusState.OK,
```

Comment:
Why the status here is still `OK`?

Reply:
Because the existing "destroy jndi-binding" did not want to report a problem if a server did not have the jndi-binding. This also looked wrong to us but we were concerned about changing the behaviour of an existing command. However, destroy jndi-binding only fails if all the servers fail. So it might be okay for us to have this return an error instead of success.

Reply:
We looked into changing the OK status from servers into ERROR to make it more clear when individual servers have issues, but this causes problems with the CliFunctionResult resolution in gfsh which disrupts behavior with the "ifExists" flag. Basically, if all the servers return the ERROR status then the gfsh code will cause the command to fail regardless of the state of 'ifExists'. 

We have code in the command which will trigger a failure for the destroy command in the event that ifExists is set to false (in which case the user expects to see their command fail if it finds nothing to delete) but will allow GFSH to report a success when ifExists is true under the same circumstances.